### ISSUE TITLE
`srb rbi update` workflow improvements

### ISSUE THREAD

**Issue by ghiculescu** [sorbet/sorbet#1930]
### Problem

I have a rake task that looks like this:

```ruby
namespace :sorbet do
  desc "Update all sorbet things."
  task :update do
    puts "Updating Sorbet based on this runbook: https://sorbet.org/docs/faq#how-do-i-upgrade-sorbet"

    puts "Updating to latest version of all Sorbet related gems we use"
    system("bundle update sorbet sorbet-rails sorbet-runtime sorbet-progress rubocop-sorbet --conservative")

    puts "Updating RBIs, https://sorbet.org/docs/rbi#quickref (takes a long time)"
    system("SRB_YES=1 bundle exec srb rbi update")

    puts "Regenerating Rails types - https://github.com/chanzuckerberg/sorbet-rails (takes a long time)"
    system("bundle exec rake rails_rbi:all")

    puts "Upgrading sigils where we can..."
    system("bundle exec srb rbi suggest-typed")

    puts "Running Sorbet. If we have errors here, something bad has happened."
    system("bundle exec srb tc")
  end
end

# bundle exec rake sorbet:update
```

The intent is to run this week periodically so that we are always on the latest sorbet, have updated rbis, etc.

The problem is that `srb rbi update` takes a *really* long time. This is because it does a bunch of things - [particularly re-suggesting types and regenerating the todo.rbi file](https://github.com/sorbet/sorbet/blob/master/gems/sorbet/bin/srb-rbi#L126..L139) - that we'll need to redo after updating `rails_rbi`s anyway.

### Proposed solution

I'd like to be able to give `srb rbi` a list of tasks to perform. And have it just perform those, in the order given, and nothing else. If I could do that, my workflow would be:

1) `bundle update` (no changes)
2) `srb rbi run config sorbet-typed gems hidden-definitions todo` (this is a new command)
3) `rake rails_rbi:all` (no changes)
4) `srb rbi run suggest-typed todo suggest-typed` (this matches the order of commands i highlighted above. i might not need to run suggest-typed twice here. point is i can change this order whenever i like.)
5) `srb tc` (no changes)

Basically, I'm proposing a `run` command that takes an arbitrary list of tasks to run.

I'm happy to build this but first want to validate that it's worth doing and is likely to get accepted.

btw. I'd love to hear how other people are keeping Sorbet up to date in their repos. It's possible this is all a dumb way of doing this.

**Comment by jez** [comment]
@ghiculescu It's already possible to specify an individual step. So the command that you want is:

```
srb rbi config && srb rbi sorbet-typed && srb rbi gems && srb rbi hidden-definitions && srb rbi todo
```

**Comment by jez** [comment]
(I'd prefer to not add a new command if it's possible to get the same feature from a bash one liner, because then we don't have worry about breaking people if we regret adding this feature in the future and wanted to remove it)

**Comment by ghiculescu** [comment]
@jez the problem with doing it that way is you don't have the same guarantees of what tasks will run. for example `srb rbi gems` will also run [suggest-typed](https://github.com/sorbet/sorbet/blob/master/gems/sorbet/bin/srb-rbi#L202).

**Comment by yf6** [comment]
I want to second this proposal, or at least suggest taking `SuggestTyped` out of the `gems` task. Having `SuggestTyped` run as part of `gems` generates all the file diffs when we are still mid updating type definitions, and it's pretty annoying to have to undo.

Unless there is an alternative workflow I'm not aware of.

**Comment by hlascelles** [comment]
Agree with @yf6 (IIUYC). I would like my typed definitions to "ratchet up" to all `typed: true` or better when I do some kind of "update" command. Step one after an "sorbet update" for me is always a (finickity) manual git revert of all the `typed` changes.

My desired workflow would be:

```
# Optionally:
bundle update

# Important, should not regress `typed: true` to `typed: false`,
# and optionally could improve `typed: false` => `typed: true` => `typed: strict`:
srb some_new_update_all_method

# Check everything, and find the (now) errors
srb tc

# Improve my code to make sure file is at least as well `typed` as before.
```

I may be "doing it wrong", but to a newcomer, it feels like the right thing to do?

**Comment by jez** [comment]
Going to close this in favor of Tapioca

https://sorbet.org/blog/2022/07/27/srb-tapioca
